### ISSUE TITLE
"Sequence" means too many things in Raku

### ISSUE THREAD

**Issue by lizmat** [Raku/problem-solving#175]
We have a `Seq` class, a `Sequence` role, and a sequence operator `...`.  Only the first two have some relation, the last one is something completely else.

I propose we:
- start referring to `...` as the *series operator*
- deprecate the `X::Sequence::` exception classes
- create similar `X::Series::` exception classes
- adapt `make test` and roast
- adapt the appropriate documentation

**Comment by AlexDaniel** [comment]
I see so many issues with `…` operator but the name ain't one. `Seq` does `Sequence`, and `…` operator is used to create Seqs. That's not too horrible (in fact I find it rather clear).

**Comment by lizmat** [comment]
A `.map` or a `.grep` are also used to create `Seq`s.  I find that argument not very convincing.

**Comment by jnthn** [comment]
> We have a Seq class, a Sequence role,

Having a general role that has a related name to a particular, common, concrete type is common. Given `Str` does `Stringy`, `Rat` does `Rational`, and `Num` does `Numeric`, it's not unusual that `Seq` would do `Sequence`, nor that the two exist.

> the last one is something completely else

Is it? It produces a `Seq`, and while there's often more convenient ways, you can probably produce any `Seq` you would wish to using it. For example, here is how to implement `map`:

```
sub my-map(&block, Iterable $source) {
    my \i = $source.iterator;
    { my \v = i.pull-one; v =:= IterationEnd ?? v !! block(v) } ...^ *<> =:= IterationEnd
}
say my-map 2 + *, 1..10
```

Similarly, a `supply` block is just one way to create a `Supply`; are you proposing we rename that too? What about colonpairs or the operator `=>`, documented as the pair constructor, which are only two of the ways to (sometimes) produce a `Pair` instance? Again, it seems there's widespread use of a pattern here: an operator or construct is often named after the type of object it produces, even if it's not the only way to produce such objects.

In summary, if this is a design problem, it's in very good company.
